### PR TITLE
#2072 - New Program Credential Type

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Added "Undergraduate Citation" to the dropdown of Program Credentials:
  image

- Update Federal Study Code Documentation:
  image

- Updated workflow:
  - Part time:
  image
  - Full time:
  image
  image
  image

### CODE REVIEW

File: sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2021-2022.bpmn [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:zeebe="http://camunda.org/schema/zeebe/1.0" xmlns:modeler="http://camunda.org/schema/modeler/1.0" xmlns:camunda="http://camunda.org/schema/1.0/bpmn" id="Definitions_0ihicdz" targetNamespace="http://bpmn.io/schema/bpmn" exporter="Camunda Modeler" exporterVersion="5.18.0" modeler:executionPlatform="Camunda Cloud" modeler:executionPlatformVersion="8.1.0" camunda:diagramRelationId="7a75e87a-6774-4e60-9a94-5dbbe3be143a">
```
Comment (dheepak-aot) [inline sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2021-2022.bpmn:2]: For the part-time assessment *.bpmn please make sure that latest changes from main(changes fromhttps://github.com/bcgov/SIMS/pull/2816) are included in them.

File: sources/packages/forms/src/form-definitions/educationprogram.json [hunk 2]
```diff
@@ -563,6 +563,10 @@
                                 "label": "Undergraduate Certificate",
                                 "value": "undergraduateCertificate"
                             },
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/educationprogram.json:565]: The value `undergraduateCitation` is missing to be added to the hidden variable `Level Of Study Codes`

image

Comment (andrepestana-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:565]: Sorry, I did that but then I messed this a bit. Thanks!

File: sources/packages/forms/src/form-definitions/educationprogram.json [hunk 3]
```diff
@@ -563,6 +563,10 @@
...
+                            {
+                                "label": "Undergraduate Citation",
+                                "value": "undergraduateCitation"
+                            },
                             {
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/educationprogram.json:570]: `onlyAvailableItems` seems to be false for this component which needs to be set true. May not be part of PR but important.

File: sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn [hunk 4]
```diff
@@ -1478,7 +1478,7 @@ or (calculatedDataPartnerFedralProvincialPDReceiptCost &gt; 0)</bpmn:conditionEx
...
+          <zeebe:output source="=if (((calculatedDataFederalAssessedNeed + calculatedDataProvincialAssessedNeed) &#62;= 1) &#10;    and (programCredentialType = &#34;undergraduateCertificate&#34;&#10;		 or programCredentialType = &#34;undergraduateCitation&#34;&#10;         or programCredentialType = &#34;graduateCertificate&#34;&#10;         or programCredentialType = &#34;undergraduateDiploma&#34;&#10;         or programCredentialType = &#34;graduateDiploma&#34;&#10;         or programCredentialType = &#34;undergraduateDegree&#34;)&#10;    and (programLength = &#34;60WeeksToLessThan2Years&#34;&#10;         or programLength = &#34;2YearsToLessThan3Years&#34;&#10;         or programLength = &#34;3YearsToLessThan4Years&#34;&#10;         or programLength = &#34;4YearsToLessThan5Years&#34;&#10;         or programLength = &#34;5YearsOrMore&#34;)&#10;    and (calculatedDataTotalFamilyIncome &#60;= dmnFullTimeAwardFamilySizeVariables.limitAwardCSGFThresholdIncome)) &#10;then true else false" target="awardEligibilityCSGF" />
```
Comment (dheepak-aot) [inline sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1481]: The story says that in Camunda, that we should follow the same logic/path as `qualifyingStudies`.

image

But for  SABC Credentia it must follow the equivalent of `undergraduate certificate`. I would like to re-validate this point with business.

If we use the `undergraduateCitation` to follow path of qualifying studies in camunda. there is no PT impact.

image

Let me know if I am missing something.

**Comment by dheepak-aot** [comment]
Good work @andrepestana-aot . Please have a look  at the comments.

File: sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1481]: @dheepak-aot `undergraduate certificate` is used for "Field of Study" and `qualifyingStudies` for the eligibility.
The only concern from the AC perspective and that it mentions "Impacts are on PT and FT." as a complementary information.
Please confirm the business the intention but the PR is following the ACs, IMO.

Comment (andrepestana-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1481]: I'm confirming with @HRAGANBC

File: sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn
Comment (andrewsignori-aot) [inline sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1]: `awardEligibilityBCAG` also has `qualifyingStudies` as a credential type. Is it missing the new `undergraduateCitation` mapping?

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, please take a look at the comments.

File: sources/packages/forms/src/form-definitions/educationprogram.json (hunk 3 above)
Comment (andrepestana-aot) [reply sources/packages/forms/src/form-definitions/educationprogram.json:570]: I changed that to true.

File: sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn (hunk 4 above)
Comment (andrepestana-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1481]: I confirmed with Carly and did the changes to workflow. Thanks!

File: sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn
Comment (andrepestana-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1]: I confirmed with @CarlyCotton and did the changes to workflow. Thanks!

File: sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2021-2022.bpmn [hunk 5]
```diff
@@ -415,7 +415,7 @@
...
       <bpmn:extensionElements>
         <zeebe:ioMapping>
-          <zeebe:output source="=if institutionType = &#34;BC Public&#34;&#10;  and (calculatedDataTotalAssessedNeed &#62;= 1)&#10;  and (calculatedDataTotalFamilyIncome &#60; dmnPartTimeAwardFamilySizeVariables.limitAwardBCAGThresholdIncome)&#10;  and (programCredentialType = &#34;undergraduateCertificate&#34;&#10;       or programCredentialType = &#34;undergraduateDiploma&#34;&#10;       or programCredentialType = &#34;undergraduateDegree&#34;)&#10;then true &#10;else false" target="awardEligibilityBCAG" />
+          <zeebe:output source="=if institutionType = &#34;BC Public&#34;&#10;  and (calculatedDataTotalAssessedNeed &#62;= 1)&#10;  and (calculatedDataTotalFamilyIncome &#60; dmnPartTimeAwardFamilySizeVariables.limitAwardBCAGThresholdIncome)&#10;  and (programCredentialType = &#34;undergraduateCertificate&#34;&#10;	   or programCredentialType = &#34;undergraduateCitation&#34;&#10;       or programCredentialType = &#34;undergraduateDiploma&#34;&#10;       or programCredentialType = &#34;undergraduateDegree&#34;)&#10;then true &#10;else false" target="awardEligibilityBCAG" />
```
[the remark this answers is not in the document]
Comment (andrepestana-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2021-2022.bpmn:418]: Confirmed and changed the code. Thanks!

File: sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn
Comment (andrepestana-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/fulltime-assessment-2021-2022.bpmn:1]: Both confirmed with Biz. Thanks!

File: sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2021-2022.bpmn (hunk 5 above)
[the remark this answers is not in the document]
Comment (andrepestana-aot) [reply sources/packages/backend/workflow/BPMN/camunda-8/parttime-assessment-2021-2022.bpmn:418]: Updated the ACs confirming with Biz.
image

**Review by dheepak-aot — APPROVED** [review]
Thanks for making the changes and reaching out to business on clarifications  @andrepestana-aot . 👍

**Review by guru-aot — APPROVED** [review]
LGTM, nice work @andrepestana-aot

**Pull request merged**